man/systemd.exec.xml

```diff
@@ -2911,6 +2911,25 @@ StandardInputData=V2XigLJyZSBubyBzdHJhbmdlcnMgdG8gbG92ZQpZb3Uga25vdyB0aGUgcnVsZX
         </para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><varname>LogFilterRegex=</varname></term>
+
+        <listitem><para>Define a regular expression to filter log messages based on the MESSAGE= field of the
+        structured message. If defined, log messages matching against the pattern will be processed by systemd-journald.
+        If the first character of the pattern is <literal>~</literal>, the effect is inverted and log messages matching
+        the pattern are discarded. If <varname>LogFilterRegex=</varname> is not defined, all log messages are processed
+        by systemd-journald.</para>
+
+        <para>Filtering is based on the unit <varname>LogFilterRegex=</varname> is defined, meaning log messages coming
+        from <citerefentry><refentrytitle>systemd</refentrytitle><manvolnum>1</manvolnum></citerefentry> about the unit
+        are not concerned. Filtered log messages won't be forwarded to a traditional syslog daemon, to the kernel log
+        buffer (kmsg), to the systemd console, or sent as wall messages to all logged-in users.</para>
+
+        <para>If a log message contains one or more NUL characters, the filter will be applied up to the first NUL
+        character. Text placed after the first NUL character in the log message won't affect the filtering
+        result.</para></listitem>
```

Comment:
hmm, the pcre code accepts a size, hence is this actually true?

Reply:
It is. The code passes `strlen(message)` to pcre instead of actual sizes: https://github.com/systemd/systemd/pull/24058/files#diff-1f15879a8745a8e8e748caac489648264f78a5e2d54d1dc0e7f44a5b7460aa55. Fishy syslog messages can't be filtered out either: https://github.com/systemd/systemd/pull/24058#discussion_r929991685

Reply:
To add more details:
- [`server_process_entry()`](https://github.com/systemd/systemd/blob/main/src/journal/journald-native.c#L95) does not provide easy access to the message size. It doesn't handle `\0` in messages at the moment.
- `server_process_syslog_message()` has `raw_len` available but this is the length of the whole message, including metadata. We don't filter against metadata.

Usage of `strlen()` and pattern matching up the `\0` is a consequence of a limitation in `journald-native.c`. I'll create an issue for it.

Reply:
Well, @evverx is right that we don't process stuff beyond NUL. But this is not a property of this new regex code so much, but off journald's processing. Hence it shouldn't really be documented here. (And not documented for journald either elsewhere, simply because we instead should just address the NUL byte thing, and allow it to be passed through, or be escape or so.)

Reply:
> But this is not a property of this new regex code so much, but off journald's processing. Hence it shouldn't really be documented here.

Fair enough. I think the comment in the code and a bug report on GitHub should be enough to kind of document this temporary limitation. It would be harder to find but it's not that the other limitation are documented either.